internal/util/porch/function.go

```diff
@@ -0,0 +1,160 @@
+		Timeout(PorchExpiration).
+		Resource("functions").
+		VersionedParams(&metav1.GetOptions{}, codec).
+		Do(f.ctx).
+		Into(&functions); err != nil {
+		return nil
+	}
+	return functions.Items
+}
+
+type Matcher interface {
+	Match(v1alpha1.Function) bool
+}
+
+var _ Matcher = TypeMatcher{}
+var _ Matcher = KeywordsMatcher{}
+
+type TypeMatcher struct {
+	FnType string
+}
+
+// Match determines whether the `function` (which can be multi-typed), belongs
+// to the expectedType. type value should only be `validator` or `mutator`.
+func (m TypeMatcher) Match(function v1alpha1.Function) bool {
+	for _, actualType := range function.Spec.FunctionTypes {
+		if string(actualType) == m.FnType {
+			return true
+		}
+	}
+	return false
+}
+
+type KeywordsMatcher struct {
+	Keywords []string
+}
+
+// Match determines whether the `function` has keywords which match the `expectedKeywords`.
+// Experimental: This logic may change to only if all function keywords are found from  expectedKeywords,
+// can it claims a match (return true).
+func (m KeywordsMatcher) Match(function v1alpha1.Function) bool {
+	if len(m.Keywords) == 0 {
+		// Accept all functions if keywords are not given.
+		return true
+	}
+	for _, actual := range function.Spec.Keywords {
+		for _, expected := range m.Keywords {
+			if actual != expected {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func MatchFunctions(ctx context.Context, matchers ...Matcher) []v1alpha1.Function {
+	var suggestedFunctions []v1alpha1.Function
+	functions := FunctionGetter{ctx: ctx}.Get()
+	for _, function := range functions {
+		for _, matcher := range matchers {
+			if matcher.Match(function) {
+				suggestedFunctions = append(suggestedFunctions, function)
+			}
+		}
+	}
+	return suggestedFunctions
+}
+
+// ToShortNames trims the function image to remove the OCI repo prefix and
+// only show the actual image and tag (or digest).
+// TODO(yuwenma): or may users prefer the Function `meta.name`?
+func ToShortNames(functions []v1alpha1.Function) []string {
+	var shortNameFunctions []string
+	for _, function := range functions {
+		slash := strings.LastIndex(function.Spec.Image, "/")
+		shortNameFunctions = append(shortNameFunctions, function.Spec.Image[slash+1:])
+	}
+	return shortNameFunctions
+}
+
+func UnifyKeywords(functions []v1alpha1.Function) []string {
+	var keywords []string
+	keywordsMap := map[string]bool{}
+	for _, function := range functions {
+		for _, kw := range function.Spec.Keywords {
+			if _, ok := keywordsMap[kw]; !ok {
+				keywordsMap[kw] = true
+				keywords = append(keywords, kw)
+			}
+		}
+	}
+	return keywords
+}
+
+func GetKeywordsFromFlag(cmd *cobra.Command) []string {
+	flagVal := cmd.Flag("keywords").Value.String()
+	flagVal = strings.TrimPrefix(flagVal, "[")
+	flagVal = strings.TrimSuffix(flagVal, "]")
+	splitted := strings.Split(flagVal, ",")
+	var trimmed []string
+	for _, val := range splitted {
+		trimmed = append(trimmed, strings.TrimSpace(val))
+	}
+	return trimmed
+}
```

Comment:
[nit] I'd recommend moving this function out to the command-related code, i.e. into cmdutil.go to keep the porch code unaware of cobra.